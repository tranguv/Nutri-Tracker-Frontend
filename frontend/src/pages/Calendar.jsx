import React from 'react'
import {DatePickerComponent} from '@syncfusion/ej2-react-calendars'
import { ScheduleComponent, ViewsDirective,ViewDirective, Day, Week, WorkWeek,Month, Agenda, Inject, Resize, DragAndDrop } from '@syncfusion/ej2-react-schedule'
const Calendar = () => {
        const data = [
                {
                    Id: 1,
                    Subject: 'Meeting - 1',
                    StartTime: new Date(2024, 1, 15, 10, 0),
                    EndTime: new Date(2018, 1, 16, 12, 30),
                    IsAllDay: false
                },
            ];
            const eventSettings = { dataSource: data }
    return (
        <div className='m-2  mt-12 p-2  bg-white rounded-3xl'>
                <ScheduleComponent height="70vh" selectedDate= {new Date(2024, 2, 15)}>
                        <Inject services={[Day, Week, WorkWeek, Month, Agenda, Resize, DragAndDrop]} />
                </ScheduleComponent>
        </div>
    )
}

export default Calendar