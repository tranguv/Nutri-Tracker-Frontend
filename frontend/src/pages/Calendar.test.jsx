import React from 'react';
import renderer from 'react-test-renderer';
import Calendar from './Calendar';

jest.mock('@syncfusion/ej2-react-calendars', () => ({
  DatePickerComponent: 'MockedDatePickerComponent'
}));

jest.mock('@syncfusion/ej2-react-schedule', () => ({
  ScheduleComponent: 'MockedScheduleComponent',
  ViewsDirective: 'MockedViewsDirective',
  ViewDirective: 'MockedViewDirective',
  Day: 'MockedDay',
  Week: 'MockedWeek',
  WorkWeek: 'MockedWorkWeek',
  Month: 'MockedMonth',
  Agenda: 'MockedAgenda',
  Inject: 'MockedInject',
  Resize: 'MockedResize',
  DragAndDrop: 'MockedDragAndDrop'
}));

describe('Calendar component', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Calendar />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
