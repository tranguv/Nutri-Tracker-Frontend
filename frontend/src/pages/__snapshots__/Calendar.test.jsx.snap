// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calendar component renders correctly 1`] = `
<div
  className="m-2  mt-12 p-2  bg-white rounded-3xl"
>
  <MockedScheduleComponent
    height="70vh"
    selectedDate={2024-03-15T04:00:00.000Z}
  >
    <MockedInject
      services={
        [
          "MockedDay",
          "MockedWeek",
          "MockedWorkWeek",
          "MockedMonth",
          "MockedAgenda",
          "MockedResize",
          "MockedDragAndDrop",
        ]
      }
    />
  </MockedScheduleComponent>
</div>
`;
